lib/build.py

```diff
@@ -832,12 +838,21 @@ def __init__(self, *args):
 class I_from_(Instruction):
 
    __slots__ = ("alias",
+                "arg",
                 "base_image")
 
    def __init__(self, *args):
       super().__init__(*args)
-      self.base_image = ch.Image(ch.Image_Ref(ch.tree_child(self.tree,
-                                                            "image_ref")))
+      image_ref = ch.Image_Ref(ch.tree_child(self.tree, "image_ref"))
+      arg_check = self.options.pop("arg", "False")
+      if (arg_check):
+         self.arg = arg_check.split("=")
+      image_ref.host = self.update_var(image_ref.host)
+      image_ref.port = self.update_var(image_ref.port)
+      image_ref.name = self.update_var(image_ref.name)
+      image_ref.tag = self.update_var(image_ref.tag)
+      image_ref.digest = self.update_var(image_ref.digest)
+      self.base_image = ch.Image(image_ref)
```

Comment:
1. There's a choice here that we ought to make explicit: that variable substitution happens (1) before or (2) after the image ref is split into parts. I think Option 1 might be quite tricky to express in the grammar, though it's more expressive.
2. Is there a way to use the existing `variables_sub` method?
3. I think you can use a loop instead of several individual calls.